src/messaging.cc

```diff
@@ -660,7 +660,8 @@ bool MessageDecoder::decode_byte_array_external(void** data, int* length) {
     return true;
   } else if (tag == TAG_BYTE_ARRAY_INLINE) {
     int encoded_length = *length = read_cardinal();
-    void* copy = malloc(encoded_length);
+    int malloc_length = encoded_length == 0 ? 1 : encoded_length;
```

Comment:
Maybe add a comment to explain why this is and maybe use `Utils::max(1, encoded_length)`? Not sure the extra `malloc_length` is worth it.